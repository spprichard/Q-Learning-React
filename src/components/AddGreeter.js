import React, {Component} from 'react'

class AddGreeter extends Component {
    constructor(props){
        super(props);
        this.handleUpdate = this.handleUpdate.bind(this);
        this.addGreeting = this.addGreeting.bind(this);
        this.state = {
            greetingName: ''
        }
    }

    addGreeting(){
        this.props.addGreeting(this.state.greetingName);
        this.setState({
            greetingName: ''
        });
    }

    handleUpdate(event){
        this.setState({
            greetingName : event.target.value

        });
    }

    render(){
        return (
            <div className = "AddGreeter"> 
                <p className="Title">Enter your name:</p>
                <input type="text" onChange={this.handleUpdate} value={this.state.greetingName}/>
                &nbsp;&nbsp;
                <button onClick={this.addGreeting}>Add</button>
            </div>
        );
    }
}

export default AddGreeter;