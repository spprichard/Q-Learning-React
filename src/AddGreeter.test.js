import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddGreeter from './components/AddGreeter';

describe(AddGreeter, () => {
    const mockAddGreeting = jest.fn();
    const component = shallow (
        <AddGreeter addGreeting={mockAddGreeting}/>
    );

    it('renders and matches our snapshot', () => {
        const component = renderer.create(
            <AddGreeter/>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("comtains form elements", () => {
        expect(component.find('input')).toHaveLength(1);
        expect(component.find('button')).toHaveLength(1);
    });



});
