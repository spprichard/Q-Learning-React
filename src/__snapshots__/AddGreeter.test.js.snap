// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class AddGreeter extends _react.Component {
    constructor(props) {
        super(props);
        this.handleUpdate = this.handleUpdate.bind(this);
        this.addGreeting = this.addGreeting.bind(this);
        this.state = {
            greetingName: '' };

    }

    addGreeting() {
        this.props.addGreeting(this.state.greetingName);
        this.setState({
            greetingName: '' });

    }

    handleUpdate(event) {
        this.setState({
            greetingName: event.target.value });


    }

    render() {
        return (
            _react2.default.createElement('div', { className: 'AddGreeter', __source: { fileName: _jsxFileName, lineNumber: 29 }, __self: this },
                _react2.default.createElement('p', { className: 'Title', __source: { fileName: _jsxFileName, lineNumber: 30 }, __self: this }, 'Enter your name:'),
                _react2.default.createElement('input', { type: 'text', onChange: this.handleUpdate, value: this.state.greetingName, __source: { fileName: _jsxFileName, lineNumber: 31 }, __self: this }), '\\xA0\\xA0',

                _react2.default.createElement('button', { className: 'AddButton', onClick: this.addGreeting, __source: { fileName: _jsxFileName, lineNumber: 33 }, __self: this }, 'Add')));


    }} renders and matches our snapshot 1`] = `
<div
  className="AddGreeter"
>
  <p
    className="Title"
  >
    Enter your name:
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
    
  <button
    className="AddButton"
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
